src/Serializer/AbstractItemNormalizer.php

```diff
@@ -159,6 +163,82 @@ protected function getAllowedAttributes($classOrObject, array $context, $attribu
      * {@inheritdoc}
      */
     protected function setAttributeValue($object, $attribute, $value, $format = null, array $context = [])
+    {
+        $this->setValue($object, $attribute, $this->prepareAttributeValue($attribute, $value, $format, $context));
+    }
+
+    protected function instantiateObject(array &$data, $class, array &$context, \ReflectionClass $reflectionClass, $allowedAttributes, string $format = null)
+    {
+        if (null !== $object = $this->extractObjectToPopulate($class, $context, static::OBJECT_TO_POPULATE)) {
+            unset($context[static::OBJECT_TO_POPULATE]);
+
+            return $object;
+        }
+
+        $constructor = $this->getConstructor($data, $class, $context, $reflectionClass, $allowedAttributes);
+        if ($constructor) {
+            $constructorParameters = $constructor->getParameters();
+
+            $params = array();
+            foreach ($constructorParameters as $constructorParameter) {
+                $paramName = $constructorParameter->name;
+                $key = $this->nameConverter ? $this->nameConverter->normalize($paramName) : $paramName;
+
+                $allowed = false === $allowedAttributes || \in_array($paramName, $allowedAttributes);
+                $ignored = !$this->isAllowedAttribute($class, $paramName, $format, $context);
+                if ($constructorParameter->isVariadic()) {
+                    if ($allowed && !$ignored && (isset($data[$key]) || array_key_exists($key, $data))) {
```

Comment:
why using both isset and array_key_exists?

Reply:
For performance I guess (`isset` is faster)